coordinator/src/main/java/org/apache/uniffle/coordinator/web/resource/ServerResource.java

```diff
@@ -121,6 +113,26 @@ public Response<Object> decommission(@PathParam("id") String serverId) {
         });
   }
 
+  @GET
+  @Path("/nodes/summary")
+  public Response<Map<String, Integer>> getNodeStatusTotal() {
```

Comment:
If we add status in the future, we need to modify this method. Is it possible to optimize it？

Reply:
> If we add status in the future, we need to modify this method. Is it possible to optimize it？

I don't quite understand your question, the interface just gets the number of ServerNodes per state.

Reply:
I mean
```java
Map<String, Integer> stringIntegerHash = Stream.concat(clusterManager.getServerList(Collections.emptySet()).stream(), clusterManager.getLostServerList().stream())
        .collect(Collectors.groupingBy(n -> n.getStatus().name(), Collectors.reducing(0, n -> 1, Integer::sum)));
````

Reply:
> I mean
> 
> ```java
> Map<String, Integer> stringIntegerHash = Stream.concat(clusterManager.getServerList(Collections.emptySet()).stream(), clusterManager.getLostServerList().stream())
>         .collect(Collectors.groupingBy(n -> n.getStatus().name(), Collectors.reducing(0, n -> 1, Integer::sum)));
> ```

The getServerList(requiredTags) method can only obtain active nodes, and cannot obtain decommissioning, decommissioned, or unhealthy servernodes

Reply:
Sorry, i misread the following line. What will happen if it is a decommissioned or decommissioning node?
https://github.com/apache/incubator-uniffle/blob/f6a8a3585bcf700318eeec900b16671af1d287e7/coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java#L148

And how do we obtain decommissioned or decommissioning node list?
https://github.com/apache/incubator-uniffle/blob/f6a8a3585bcf700318eeec900b16671af1d287e7/coordinator/src/main/java/org/apache/uniffle/coordinator/web/resource/ServerResource.java#L68-L79

Reply:
This logic is changed .  Please take a look at #1174.

Reply:
> This logic is changed . Please take a look at #1174.

Modified merge.